### PR TITLE
Exclude dev dependencies from packages.config

### PR DESCRIPTION
**Pull request by devmehtasynopsys**
This PR will add support for excluding dev dependencies from packages.config file. This will go as part of ticket IDETECT-4140 along with the other file supports already added for 9.4.0.

### CODE REVIEW

File: detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/PackagesConfig/PackagesConfigResolver.cs [hunk 1]
```diff
@@ -55,9 +60,14 @@ private List<NugetDependency> GetDependencies()
                 var versionRange = new NuGet.Versioning.VersionRange(version, true, version, true);
                 var framework = NuGet.Frameworks.NuGetFramework.Parse(packageRef.TargetFramework.Framework);
 
+                bool excludeDevDependency = isDevDependency && packageRef.IsDevelopmentDependency;
+
                 //TODO: Check that this works.
```
Comment (andrian-synopsys) [inline detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/PackagesConfig/PackagesConfigResolver.cs:65]: While we're here, if we know that this works, we should probably remove this comment. :)

File: detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/PackagesConfig/PackagesConfigResolver.cs [hunk 2]
```diff
@@ -45,6 +48,8 @@ private List<NugetDependency> GetDependencies()
             System.IO.Stream stream = new System.IO.FileStream(PackagesConfigPath, System.IO.FileMode.Open, System.IO.FileAccess.Read);
             PackagesConfigReader reader = new PackagesConfigReader(stream);
             List<PackageReference> packages = reader.GetPackages().ToList();
+            
+            bool isDevDependency = ExcludedDependencyTypeUtil.isDependencyTypeExcluded(ExcludedDependencyTypes,"DEV");
```
Comment (andrian-synopsys) [inline detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/PackagesConfig/PackagesConfigResolver.cs:52]: The variable name seems inaccurate.

Comment (devmehtasynopsys) [reply detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/PackagesConfig/PackagesConfigResolver.cs:52]: We have used the same variable name at every place for this support previously. :)

Comment (andrian-synopsys) [reply detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/PackagesConfig/PackagesConfigResolver.cs:52]: True, and I now see that I even approved PRs where those variables were introduced in this manner.
I still vote to change it at least in this file but ideally in other files as well.

Comment (devmehtasynopsys) [reply detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/PackagesConfig/PackagesConfigResolver.cs]: Done

**Review by andrian-synopsys — APPROVED** [review]

File: detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/PackagesConfig/PackagesConfigResolver.cs [hunk 3]
```diff
@@ -45,6 +48,8 @@ private List<NugetDependency> GetDependencies()
             System.IO.Stream stream = new System.IO.FileStream(PackagesConfigPath, System.IO.FileMode.Open, System.IO.FileAccess.Read);
             PackagesConfigReader reader = new PackagesConfigReader(stream);
             List<PackageReference> packages = reader.GetPackages().ToList();
+            
+            bool isDevDependencyTypeExcluded = ExcludedDependencyTypeUtil.isDependencyTypeExcluded(ExcludedDependencyTypes,"DEV");
```
Comment (gopannabd) [inline detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/PackagesConfig/PackagesConfigResolver.cs:52]: Is it case sensitive?

**Review by gopannabd — APPROVED** [review]

File: detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/PackagesConfig/PackagesConfigResolver.cs (hunk 3 above)
Comment (devmehtasynopsys) [reply detect-nuget-inspector/detect-nuget-inspector/DependencyResolution/PackagesConfig/PackagesConfigResolver.cs:52]: Yes it is case sensitive.

**Pull request merged**